File: cpp_solutions/94. Binary Tree Inorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
// left , root.val , right
class Solution {
public:
    vector<int> ans;
    vector<int> inorderTraversal(TreeNode* root) {
        inorder(root);
        return ans;
    }
    
    void inorder(TreeNode* root){
        
        if (root){
            if (root->left){
            inorder(root->left);
        }
            ans.push_back(root->val);
            
            if (root->right){
                inorder(root->right);
            }
        }
        
        
    }
};

File: python3_solution_set0/94. Binary Tree Inorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        inorder = self.inorder(root,ans)
        return inorder
        
        
        
    def inorder(self,root,ans):
        
        if root:
            self.inorder(root.left,ans)
            
            ans.append(root.val)
        
            self.inorder(root.right,ans)
            
        return ans
            
        
